### ISSUE TITLE
JDK 17.0.9+9 crashed due to [libjvm.so+0x2dd5f4]  unnecessary_acquire(Node const*)+0x40

### ISSUE THREAD

**Issue by andreezy777** [adoptium/adoptium-support#1004]
### Please provide a brief summary of the bug

Application crashing every +- 24 hrs.
Unfortunately it was running on a container in k8s which was restarted after the crash so I can't access the error report files.
Will try to receive it somehow.

Previously it successfully worked on 11.0.19+7.

Also I find out that you released 17.0.10 but i don't see fix for this bug.

### Did you test with the latest update version?

```Shell
no
```

### Please provide steps to reproduce where possible

run app on 17.0.9+9 jdk version, that's it.

### Expected Results

app will not crash

### Actual Results

crashed

### What Java Version are you using?

openjdk 17.0.9 2023-10-17 OpenJDK Runtime Environment Temurin-17.0.9+9 (build 17.0.9+9) OpenJDK 64-Bit Server VM Temurin-17.0.9+9 (build 17.0.9+9, mixed mode, sharing)

### What is your operating system and platform?

using `eclipse-temurin:17-jre-jammy` with linux/aarch64

### Did you test with other Java versions?

```Shell
openjdk 11.0.19 2023-04-18
OpenJDK Runtime Environment Temurin-11.0.19+7 (build 11.0.19+7)
OpenJDK 64-Bit Server VM Temurin-11.0.19+7 (build 11.0.19+7, mixed mode, sharing)
```

### Relevant log output

```Shell
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x0000ffffbd06d5f4, pid=1, tid=16
#
# JRE version: OpenJDK Runtime Environment Temurin-17.0.9+9 (17.0.9+9) (build 17.0.9+9)
# Java VM: OpenJDK 64-Bit Server VM Temurin-17.0.9+9 (17.0.9+9, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, shenandoah gc, linux-aarch64)
# Problematic frame:
# V  [libjvm.so+0x2dd5f4]  unnecessary_acquire(Node const*)+0x40
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/share/apport/apport -p%p -s%s -c%c -d%d -P%P -u%u -g%g -- %E" (or dumping to //core.1)
#
# An error report file with more information is saved as:
# /tmp/hs_err_pid1.log
#
# Compiler replay data is saved as:
# /tmp/replay_pid1.log
#
# If you would like to submit a bug report, please visit:
#   https://github.com/adoptium/adoptium-support/issues
#

[error occurred during error reporting (), id 0x5, SIGTRAP (0x5) at pc=0x0000ffffbe0b71ec]
```

**Comment by karianna** [comment]
@andreezy777 We will require a 17.0.10 crash log (code paths can change) are you able to upgrade and send us one in?(assuming it still crashes)

**Comment by andreezy777** [comment]
@karianna it'll a lit bit hard, because we faced with this issue on prod env, so we decided to rollback on 11 version.
I'll try to find out ways how to reproduce it on dev env with 17.0.9 and then will try to upgrade to 17.0.10 and will come back.

**Comment by andreezy777** [comment]
and looks like `Eclipse Adoptium, OpenJDK Runtime Environment, 21.0.1+12-LTS` works fine for more than 26 hours, 
for example on 17.0.10 our test application crashed 5 times during 15 hours.

**Comment by karianna** [comment]
@andreezy777 Thanks, if you can uplaod the full log(s) that would be ace!

**Comment by andreezy777** [comment]
[hs_err_pid1.log](https://github.com/adoptium/adoptium-support/files/14151036/hs_err_pid1.log)
[replay_pid1.log](https://github.com/adoptium/adoptium-support/files/14151037/replay_pid1.log)

@karianna I managed to get the logs

**Comment by varikin** [comment]
I have found a very similar error. I can reproduce it on multiple Debian 10 VMs on an M1 mac, so aarch64. I cannot reproduce it on x86_64 architecture. I originally saw this with 17.0.7+7 and then reproduced on 17.0.10+7 as requested above. I haven't tried 21 or another major version. Here's my logs:

[hs_err_pid19300.log](https://github.com/adoptium/adoptium-support/files/14170946/hs_err_pid19300.log)
[replay_pid19300.log](https://github.com/adoptium/adoptium-support/files/14170947/replay_pid19300.log)

**Comment by karianna** [comment]
https://bugs.openjdk.org/browse/JDK-8325372

**Comment by varikin** [comment]
I saw they asked about reproducing this. For us, this is a private company application so it will be very hard to share. Since this to be related to Apache HTTP client, this is the version we use: `org.apache.httpcomponents.client5:httpclient5:5.1.4`.

I can get more data if needed. We have an integration test suite that causes this, so it's easy to reproduce.

I'd add this upstream, but I don't have an account.

**Comment by andreezy777** [comment]
@karianna 
Sorry, I've missed messages.
We are also private company, so I can't share all classes.

If it will help:
we are using `io.opentelemetry.javaagent:opentelemetry-javaagent:1.23.0` which has dependency on `com.squareup.okhttp3:okhttp:4.10.0`

**Comment by andreezy777** [comment]
@karianna hi, I see good news. Issue was reproduced and fixed.

Can you let us know when we can expect a container with this fix?

**Comment by karianna** [comment]
At the moment the fix version is set to Java 23 with no backports as of yet.  If you have a support contract with an Adoptium member, they may be able to expedite that for you.

**Comment by andreezy777** [comment]
@karianna 
Why won't there be a backport? 
I saw that similar issues were backported to 17 version of java.
Or it will be backported but later? Do you know?

**Comment by karianna** [comment]
@andreezy777 It's down to the choice of folks who maintain OpenjDK.  Typically if it's an issue that impacts a wide range of folks one of the folks working for an OpenJDK vendor fixes it.  Else it's up to a community volunteer or you can pay for a commercial support agreement

**Comment by andreezy777** [comment]
@karianna 
> community volunteer

you mean someone from openjdk developer team or me?

**Comment by karianna** [comment]
> @karianna
> 
> > community volunteer
> 
> you mean someone from openjdk developer team or me?

Sometimes an OpenJDK Author/Committer who works for one of the vendors resolves it, sometimes it's an independent developer or academic. If you have a commercial support agreement then you can generally ask that vendor to go fix.

**Comment by timothyg-stripe** [comment]
@andreezy777 Looks like Dan Lutker from Amazon has backported the fix to JDK 17, so barring anything unexpected it should be fixed in Temurin 17.0.12. https://bugs.openjdk.org/browse/JDK-8325372
